apps/remix-ide/src/blockchain/execution-context.js

```diff
@@ -189,14 +188,16 @@ export class ExecutionContext {
     const oldProvider = web3.currentProvider
 
     web3.setProvider(endpoint)
-
     web3.eth.net.isListening((err, isConnected) => {
-      if (!err && isConnected) {
+      if (!err && isConnected === true) {
         this.executionContext = context
         this._updateBlockGasLimit()
         this.event.trigger('contextChanged', [context])
         this.event.trigger('web3EndpointChanged')
         cb()
+      } else if (isConnected === 'canceled') {
+        web3.setProvider(oldProvider)
+        cb()
```

Comment:
yes, indeed that's a clever solution.
Perhaps `cb()` should also return an `error` (same as in the `else` below)

Reply:
Generally, no message is shown on clicking on `cancel` in a modal